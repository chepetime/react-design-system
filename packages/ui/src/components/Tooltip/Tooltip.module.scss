@use "./../../styles" as styles;

:root {
  --tooltip-text-color: white;
  --tooltip-background-color: black;
  --tooltip-margin: 30px;
  --tooltip-arrow-size: 6px;
}

:global :root .dark {
  --tooltip-text-color: black;
  --tooltip-background-color: white;
  --tooltip-margin: 30px;
  --tooltip-arrow-size: 6px;
}

.Tooltip {
  @include styles.styles-reset;
  display: inline-block;
  position: relative;
  font-size: inherit;
}

/* Absolute positioning */
.Tooltip-Tip {
  position: absolute;
  border-radius: 4px;
  left: 50%;
  transform: translateX(-50%);
  padding: 6px;
  color: var(--tooltip-text-color);
  background: var(--tooltip-background-color);
  font-size: 14px;
  font-family: sans-serif;
  line-height: 1;
  z-index: 100;
  white-space: nowrap;

  /* CSS border triangles */
  &::before {
    content: " ";
    left: 50%;
    border: solid transparent;
    height: 0;
    width: 0;
    position: absolute;
    pointer-events: none;
    border-width: var(--tooltip-arrow-size);
    margin-left: calc(var(--tooltip-arrow-size) * -1);
  }
}

/* Absolute positioning */
.top {
  top: calc(var(--tooltip-margin) * -1);
}

/* CSS border triangles */
.top::before {
  top: 100%;
  border-top-color: var(--tooltip-background-color);
}

/* Absolute positioning */
.right {
  left: calc(100% + var(--tooltip-margin));
  top: 50%;
  transform: translateX(0) translateY(-50%);
}

/* CSS border triangles */
.right::before {
  left: calc(var(--tooltip-arrow-size) * -1);
  top: 50%;
  transform: translateX(0) translateY(-50%);
  border-right-color: var(--tooltip-background-color);
}

/* Absolute positioning */
.bottom {
  bottom: calc(var(--tooltip-margin) * -1);
}

/* CSS border triangles */
.bottom::before {
  bottom: 100%;
  border-bottom-color: var(--tooltip-background-color);
}

/* Absolute positioning */
.left {
  left: auto;
  right: calc(100% + var(--tooltip-margin));
  top: 50%;
  transform: translateX(0) translateY(-50%);
}

/* CSS border triangles */
.left::before {
  left: auto;
  right: calc(var(--tooltip-arrow-size) * -2);
  top: 50%;
  transform: translateX(0) translateY(-50%);
  border-left-color: var(--tooltip-background-color);
}
