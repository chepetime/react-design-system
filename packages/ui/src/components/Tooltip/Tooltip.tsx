import React from "react";
import classnames from "classnames";
import css from "./Tooltip.module.scss";

export interface TooltipProps extends React.ComponentPropsWithRef<"div"> {
  children?: React.ReactNode;
}

export const Tooltip = React.forwardRef<HTMLDivElement, TooltipProps>(
  (props: TooltipProps, ref: React.Ref<HTMLDivElement>) => {
    return (
      <div className={classnames(css.Tooltip)} ref={ref}>
        <p>Tooltip component working!</p>
        <div>{props.children}</div>
      </div>
    );
  }
)

Tooltip.displayName = "Tooltip";
