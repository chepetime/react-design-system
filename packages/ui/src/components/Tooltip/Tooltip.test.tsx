
import React from "react";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom";
import { Tooltip } from "./Tooltip";
import { render, screen } from "@testing-library/react";

describe("Tooltip", () => {
  it("renders without errors", () => {
    render(<Tooltip />);
    // Assert that the component renders without throwing any errors
  });

  it("renders with children", () => {
    render(<Tooltip>Some children</Tooltip>);
    // Assert that the component renders the provided children
    const childrenElement = screen.getByText("Some children");
    expect(childrenElement).toBeInTheDocument();
  });

  it("handles user interactions", () => {
    render(<Tooltip>Some children</Tooltip>);
    // Simulate user interactions with the component
    userEvent.click(screen.getByText("Some children"));
    // Assert the expected outcome of the user interaction
    // For example, check if a certain state has changed or an event handler has been called
  });

  it("matches snapshot", () => {
    const { container } = render(<Tooltip />);
    // Assert that the rendered component matches a previously captured snapshot
    expect(container).toMatchSnapshot();
  });
});
