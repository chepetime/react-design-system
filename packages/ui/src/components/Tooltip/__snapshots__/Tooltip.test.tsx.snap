// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip matches snapshot 1`] = `
<div>
  <div
    class="Tooltip"
  >
    <p>
      Tooltip component working!
    </p>
    <div />
  </div>
</div>
`;
